api/controllers/album.js

```diff
@@ -1,13 +1,44 @@
 // function: allAlbums
 // sends all parent albums
-export async function allAlbums(req, res) {
+
+import { query } from 'express';
+
+/**
+ * /albums : all albums
+ * /albums?nameincl={name} : Search for album that matches this name (fuzzy match)
+ * /albums?tag={tag} : Search for albums that have these tags (fuzzy matches)
+ * /albums?keyword={keyword} : Search for albums that have these keywords (fuzzy matches)
+ * /albums?parentid={parentid} : Search for albums that has this albumId as a parent.
+ * /albums?updatedafter={date} : Search for albums that were last updated after this date.
+ */
+export async function albumsBy(req, res) {
   let conn;
+  const nameIncludes = req.query.nameincl;
+  // lint/prettier keeps trying to {tag} but everything breaks then
+  //const { tag } = req.query;
+
+  const tag = req.query.tag;
```

Comment:
the params that are strings could be sanitized. somethings like 
- check if its a string,  
- if the string length is greater than 0. 
- trim whitespace from the ends of the string
- Convert case ? 

(@ManjotS can a `WHERE` clause that uses `LIKE` be case in-sensitive)?